filebeat/testing/integration/integration.go

```diff
@@ -69,8 +70,14 @@ type test struct {
 func (fbt *test) ExpectEOF(files ...string) Test {
 	// Ensuring we completely ingest every file
 	for _, filename := range files {
-		line := fmt.Sprintf("End of file reached: %s; Backoff now.", filename)
-		fbt.ExpectOutput(line)
+		// accounts for reaching EOF, but still ingesting if more data is added
+		// to the file.
+		line1 := fmt.Sprintf("End of file reached: %s; Backoff now.", filename)
+		// accounts for closing the file on EOF.
+		line2 := fmt.Sprintf("EOF has been reached. Closing. Path='%s'", filename)
+
+		fbt.ExpectOutputRegex(regexp.MustCompile(
+			fmt.Sprintf(`(%s|%s)`, line1, line2)))
```

Comment:
This would not work because the lines might contain characters that need to be escaped in regular expressions.